Multisig/Data/Services/Safe Transaction Service/SignTransactionRequest.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SignTransactionRequest.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 17.09.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Web3
+
+struct SignTransactionRequest: JSONRequest {
+    let safe: String
+
+    let to: String?
+    let value: String
+    let data: String?
+    let operation: Int
+    let safeTxGas: String
+    let baseGas: String
+    let gasPrice: String
+    let gasToken: String?
+    let refundReceiver: String
+    let nonce: String
+    let contractTransactionHash: String
+    let sender: String
+    let signature: String?
+
+    enum CodingKeys: String, CodingKey {
+        case sender, to, nonce, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver,
+             contractTransactionHash, data, operation, value, signature
+    }
+
+    enum RequestError: String, Error {
+        case couldNotVerifySafeTxHash
+    }
+
+    init(transaction: Transaction, safeAddress: Address) throws {
+        safe = safeAddress.checksummed
+        to = transaction.to?.description
+        value = transaction.value.description
+        data = transaction.data?.description
+        operation = transaction.operation.rawValue
+        safeTxGas = transaction.safeTxGas.description
+        baseGas = transaction.baseGas.description
+        gasPrice = transaction.gasPrice.description
+        gasToken = transaction.gasToken?.description
+        refundReceiver = transaction.refundReceiver.description
+        nonce = transaction.nonce.description
+        contractTransactionHash = transaction.safeTxHash.description
+
+        let hashToSign = Data(ethHex: contractTransactionHash)
+
+        guard let pkData = try? App.shared.keychainService.data(forKey: KeychainKey.ownerPrivateKey.rawValue),
```

Comment:
please use the `privateKey` in full name, and `signature`, too

Reply:
ok

Reply:
signature will be in conflict with struct var, so I will leave sig here

Reply:
To resolve the conflict, the code could be `self.signature`

Reply:
ok. will change in the next pr